sops/__init__.py

```diff
@@ -202,10 +207,17 @@ def main():
         else:
             print("%s doesn't exist, creating it." % args.file)
 
+    if args.unencrypted_suffix:
+        unencrypted_suffix = args.unencrypted_suffix
+    elif 'sops' in tree and 'unencrypted_suffix' in tree['sops']:
+        unencrypted_suffix = tree['sops']['unencrypted_suffix']
+    else:
+        unencrypted_suffix = DEFAULT_UNENCRYPTED_SUFFIX
+
     if args.encrypt:
         # Encrypt mode: encrypt, display and exit
         key, tree = get_key(tree, need_key)
-        tree = walk_and_encrypt(tree, key)
+        tree = walk_and_encrypt(tree, key, unencrypted_suffix=unencrypted_suffix)
```

Comment:
That feels pretty dirty, I mentioned this in the initial PR notes but maybe a classical structure for trees would be better (e.g. save `unencrypted_suffix`, `digest`, and `iv` to `self` instead of passing them around)?